cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -937,7 +986,65 @@ public CloudServiceInstance getServiceInstance(String serviceName, boolean requi
 
     @Override
     public List<CloudService> getServices() {
-        return fetchListWithAuxiliaryContent(this::getServiceInstanceResources, this::zipWithAuxiliaryServiceContent);
+        final List<CloudService> cloudServices = fetchListWithAuxiliaryContent(this::getServiceInstanceResources,
+                                                                               this::zipWithAuxiliaryServiceContent);
+        final Map<String, Metadata> serviceInstancesMetadata = getServiceInstancesMetadata(cloudServices.stream()
+                                                                                                        .map(service -> service.getName())
+                                                                                                        .collect(Collectors.toList()));
```

Comment:
Extract this to method or check whether there is a function which maps services to their names